hack/generator/pkg/astmodel/type_definition.go

```diff
@@ -68,6 +76,19 @@ func (def TypeDefinition) AsDeclarations(codeGenerationContext *CodeGenerationCo
 	return def.theType.AsDeclarations(codeGenerationContext, def.name, def.description)
 }
 
+// AddFlag includes the specified flag on the resource
+func (def TypeDefinition) AddFlag(flag Flag) TypeDefinition {
+	result := def.copy()
+	result.flags[flag] = struct{}{}
+	return result
+}
+
+// HasFlag returns true if the specified flag is present on this resource
+func (def TypeDefinition) HasFlag(flag Flag) bool {
+	_, ok := def.flags[flag]
```

Comment:
Slightly awkward here as I mentioned above since this `TypeDefinition` can be of a `Type` (`Resource`) that can itself have Flags.

As a caller that feels awkward because then doesn't my check end up something like:
```
func someFunc(types astmodel.Types) (astmodel.Types, error) {
    for typeName, def := range types {
        hasFlag := false
        asResource, ok := def.Type().(*astmodel.Resource)
        if ok {
            hasFlag = asResource.HasFlag("foo")
        }
        hasFlag |= def.HasFlag("foo")
}
```

Reply:
No, because you know where you added the flag originally. If you added the flag to an `ObjectType`, that's where you check; if you added it to a `Resource`, that's where  you check. 

Flags are a coordination primitive - I'd expect the creator and consumer of a flag to be written by the same person at the same time.